File: src/DataStructuresImplementing/list/_DynamicList.java
package DataStructuresImplementing.list;

import java.util.Arrays;

public class _DynamicList<T> {
        Object[] data;
        int size;
public _DynamicList() {
        size=0;
        data=new Object[1];
        }
public int getSize(){
        return data.length;
        }
public T get(int index){
        return (T) data[index];
        }
public void put(Object element){
        ensureCapacity(size+1);
        data[size++]=element;
        }

public void ensureCapacity( int minCapacity){
        int oldCapacity= getSize();
        if(minCapacity>oldCapacity){
        int newCapacity=oldCapacity*2;
        if(newCapacity<minCapacity)
        newCapacity=minCapacity;
        data= Arrays.copyOf(data, newCapacity);
        }
        }
        }

File: src/DataStructuresImplementing/list/_DynamicList.kt
package DataStructuresImplementing.list

/*
class DynamicList<T> {
    var data: Array<Any?>
    var size: Int = 0

    init {
        data = arrayOfNulls(1)
    }

    fun getSize(): Int {
        return data.size
    }

    fun get(index: Int): T? {
        return data[index] as T?
    }

    fun put(element: Any?) {
        ensureCapacity(size + 1)
        data[size++] = element
    }

    fun ensureCapacity(minCapacity: Int) {
        val oldCapacity = getSize()
        if (minCapacity > oldCapacity) {
            var newCapacity = oldCapacity * 2
            if (newCapacity < minCapacity) newCapacity = minCapacity
            data = data.copyOf(newCapacity)
        }
    }
}*/